public/app/plugins/datasource/grafana/utils.ts

```diff
@@ -54,3 +57,16 @@ function updateSnapshotData(frames: DataFrame[], panel: PanelModel) {
 
   panel.refresh();
 }
+
+export function fileDropHandler(onFileLoaded: (query: GrafanaQuery, fileRejections: FileRejection[]) => void) {
```

Comment:
This function is not the fileDropHandler itself. It produces the handler so I'd do a small renaming here to `getFileDropToQueryHandler`.